vnet/lib/vnet/endpoints/1.0/topologies.rb

```diff
@@ -45,11 +45,19 @@ def self.put_post_shared_params
     assoc_response = R.const_get("Topology#{other_name.camelize}")
     
     # TODO: Change to confirm with either POST or PUT idioms.
-    param_uuid M::Interface, :interface_uuid, required: true if with_interface
+    if with_interface
+      param_uuid M::Interface, :interface_uuid, required: true
+      param_uuid M::IpLease, :ip_lease_uuid, required: false
```

Comment:
This needs a test added for this new parameter here: https://github.com/axsh/openvnet/blob/fix-toplogy/vnet/spec/vnet/endpoints/1.0/topologies_spec.rb#L76

Vnctl also needs to be updated: https://github.com/axsh/openvnet/blob/fix-toplogy/client/vnctl/lib/vnctl/cli/topology.rb#L14-L17

Actually I was surprised that the integration test passed without this parameter being in vnctl. I checked the dataset and noticed that this parameter isn't in there. Is it being tested at all?

Also just out of interest, I was wondering how in what cases this parameter is to be used and in what cases it can be omitted.

Reply:
The ip_lease_uuid parameter is optional if the interface, in which case it selects the first ip lease on the interface.

Reply:
Fixed.